File: Python/747. Largest Number At Least Twice of Others.py
class Solution:
    def dominantIndex(self, nums: List[int]) -> int:
        temp = list(nums)
        temp.sort(reverse = True)
        if len(temp) == 1: return 0
        if temp[0] >= temp[1] * 2: return nums.index(temp[0])
        else: return -1

File: C++/747. Largest Number At Least Twice of Others.cpp
class Solution {
public:
    int dominantIndex(vector<int>& nums) {
        ios_base::sync_with_stdio(false);
        cin.tie(NULL);
        cout.tie(NULL);

        int res, temp = 0;

        for (int i = 0; i < nums.size(); i++) {
            if (nums[i] > temp) {
                res = i;
                temp = nums[i];
            }
        }

        nums[res] = 0;

        for (int i: nums) {
            if (i * 2 > temp) return -1;
        }

        return res;
    }
};
